holmes.go

```diff
@@ -667,3 +698,54 @@ func (h *Holmes) Set(opts ...Option) error {
 	}
 	return nil
 }
+
+func (h *Holmes) DisableProfileReporter() {
+	atomic.StoreInt32(&h.opts.pReportOpts.active, 0)
+}
+
+func (h *Holmes) ReportProfile(pType string, buf []byte, reason string, eventID string) {
+	opts := h.opts.GetReporterOpts()
+	if opts.active == 0 {
+		return
+	}
+	h.opts.pReportOpts.eventsCh <- rptEvent{
+		pType,
+		buf,
+		reason,
+		eventID}
+}
+
+// startReporter starts a background goroutine to consume event channel,
+// and finish it at after receive from cancel channel.
+func (h *Holmes) startReporter() {
+	// opts.reporter is nil for now because we haven't call WithProfileReporter to initial it yet.
+	opts := h.opts.GetReporterOpts()
+	go func(cancelCh <-chan struct{}, eventsCh <-chan rptEvent, changeReporterCh <-chan ProfileReporter) {
```

Comment:
I won't pass `h` or `opts` into background gr function, so I added a channel `changeReporterCh ` to notify background gr.

Reply:
I don't think this is a good idea.
we should always fetch the reporter interface every time before we use it (inside the background goroutine).